app/services/menu/loader.rb

```diff
@@ -74,12 +74,14 @@ def self.load
         menu.sub_menu :configure_menu,  :caption => N_('Configure') do
           menu.item :hostgroups,        :caption => N_('Host groups')
           menu.item :common_parameters, :caption => N_('Global parameters')
-          menu.divider                  :caption => N_('Puppet')
-          menu.item :environments,      :caption => N_('Environments')
-          menu.item :puppetclasses,     :caption => N_('Classes')
-          menu.item :config_groups,     :caption => N_('Config groups')
-          menu.item :variable_lookup_keys, :caption => N_('Smart variables')
-          menu.item :puppetclass_lookup_keys, :caption => N_('Smart class parameters')
+          if Setting[:puppet_enabled]
```

Comment:
This appears to cause complete test failures because it assumes the database is available. It may be better to add support for conditionals via lambdas or similar so it can be checked during rendering?